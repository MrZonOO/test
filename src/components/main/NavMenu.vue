<template>
  <div class="nva_memu_list">
    <vue-perfect-scrollbar class="overflow_container">
      <v-list dense>
        <v-list dark class="transparent">
          <v-list-tile
            v-for="(item,i) in menus"
            :key="i"
            :to="item.to"
          >
            <v-list-tile-avatar>
              <v-icon dark>{{ item.icon }}</v-icon>
            </v-list-tile-avatar>
            <v-list-tile-content>
              <v-list-tile-title class="body-2">
                {{ item.title }}
              </v-list-tile-title>
            </v-list-tile-content>
          </v-list-tile>
        </v-list>
      </v-list>
    </vue-perfect-scrollbar>
  </div>
</template>

<script>
export default {
  props: {
    menus: {
      type: Array,
    },
  },
};
</script>


<style lang="scss">
.nva_memu_list {
  height: calc(100% - 240px);
  position: relative;
}
</style>
