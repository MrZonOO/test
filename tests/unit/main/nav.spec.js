import { shallowMount } from '@vue/test-utils';
import NavMenu from '@/components/main/NavMenu.vue';

describe('NavMenu.vue', () => {
  it('renders menu for each menus in props.menus', () => {
    const menus = [{
      title: 'Dashboard',
      icon: 'dashboard',
      to: '/dashboard',
    }, {
      title: 'Client Waiting List',
      icon: 'people',
      to: '/queue',
    }];
    const wrapper = shallowMount(NavMenu, {
      propsData: { menus: menus },
    });
    expect(wrapper.findAll('v-list-tile')).toHaveLength(menus.length);
  })

  it('matches snapshot', () => {
    const menus = [{
      title: 'Dashboard',
      icon: 'dashboard',
      to: '/dashboard',
    }, {
      title: 'Client Waiting List',
      icon: 'people',
      to: '/queue',
    }];
    const wrapper = shallowMount(NavMenu, {
      propsData: { menus: menus },
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
});
