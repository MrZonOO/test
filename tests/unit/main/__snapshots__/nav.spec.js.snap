// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavMenu.vue matches snapshot 1`] = `
<div class="nva_memu_list">
  <vue-perfect-scrollbar class="overflow_container">
    <v-list dense="">
      <v-list dark="" class="transparent">
        <v-list-tile to="/dashboard">
          <v-list-tile-avatar>
            <v-icon dark="">dashboard</v-icon>
          </v-list-tile-avatar>
          <v-list-tile-content>
            <v-list-tile-title class="body-2">
              Dashboard
            </v-list-tile-title>
          </v-list-tile-content>
        </v-list-tile>
        <v-list-tile to="/queue">
          <v-list-tile-avatar>
            <v-icon dark="">people</v-icon>
          </v-list-tile-avatar>
          <v-list-tile-content>
            <v-list-tile-title class="body-2">
              Client Waiting List
            </v-list-tile-title>
          </v-list-tile-content>
        </v-list-tile>
      </v-list>
    </v-list>
  </vue-perfect-scrollbar>
</div>
`;
